### ISSUE TITLE
Allow setting env-variable for kube-scheduler

### ISSUE THREAD

**Issue by hardikdr** [gardener/gardener#2343]
**What would you like to be added**:  
We need to allow end-users to configure the maximum number of the volumes that can be attached to the node via shoot-resource. This can be achieved by setting the `KUBE_MAX_PD_VOLS` env-variable in kube-scheduler deployment, refer the [documentation](https://kubernetes.io/docs/concepts/storage/storage-limits/#custom-limits) for more details.

**Why is this needed**:
This is specifically useful for the integration of MemoryOne based machines in gardener, where our stakeholder is willing to overwrite the maximum attachable volumes to "30" for c5d.metal machines. 

**How to categorize this issue?**

/area usability
/kind enhancement
/priority normal

**Comment by rfranzke** [comment]
> CSI drivers might have a different procedure, see their documentation on how to customize their limits.

Do you have further information? How will it work?

**Comment by hardikdr** [comment]
With initial investigation, the following are the major findings:
- For k8s version 1.15: Setting the env-var `KUBE_MAX_PD_VOLS` for kube-scheduler and having feature-gate `AttachVolumeLimit` set to `false` , allows overwriting the existing default value of 25. 
   - We dint test, but from the code-path this should also work for the k8s version 1.16.

- For k8s-version 1.17:  The solution above doesn't really seem to be effective, as the feature-gate the `AttachVolumeLimit` is disabled. It'd mean, we don't have a good way to overwrite the max-attachable-volumes.

- For k8s-version 1.18+, also with CSI-drivers enabled: We found, setting the `KUBE_MAX_PD_VOLS` has no effect at all on overwriting the max-attachable volumes. The CSI-drivers also doesn't seem to have an elegant solution available for the configuration. Follow the [conversation](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/347) for more info. 

The bottom line here is, even if we implement the solution proposed above, it might not really be useful for the versions starting from 1.17+, we need to investigate and find a better long-term solution.

**Comment by rfranzke** [comment]
/assign

As discussed, we will allow modifying the `KUBE_MAX_PD_VOLS` variable on end-user's own risk (they should be aware of side-effects and consequences, we consider it alpha-level similar to feature gates or runtime config).

The environment variable has only effect for shoots not using CSI. For example, AWS uses CSI beginning with 1.18, so here we'll be following a different approach.
